src/engine/strat_engine/pool.rs

```diff
@@ -448,10 +448,18 @@ impl Pool for StratPool {
     fn destroy_filesystems<'a, 'b>(&'a mut self,
                                    fs_uuids: &[&'b FilesystemUuid])
                                    -> EngineResult<Vec<&'b FilesystemUuid>> {
-        for fsid in fs_uuids {
-            try!(self.mdv.rm_fs(fsid));
+        let dm = try!(DM::new());
+
+        let mut removed = Vec::new();
+        for uuid in fs_uuids {
+            if let Some(fs) = self.filesystems.remove_by_uuid(&uuid) {
+                try!(self.mdv.rm_fs(uuid));
+                try!(fs.destroy(&dm, &self.thin_pool.thin_pool));
```

Comment:
This is very iffy. We should consider two alternatives:

1. Storing filesystems in ThinPool.
2. Storing thindevs in ThinPool and storing only thindev ids in StratFilesystem.